File: ronit/c/star_pattern.c
// Online C compiler to run C program online
#include <stdio.h>

int main() {
    // Write C code here
    int run_again = 0;
    int row = 0,col = 0;
    do
    {
        printf("enter the number of rows in triangle\r\n");
        scanf("%d", &row);
        
    /* Right inclined left angle triangle star pattern  */
        for(int row_loop=0;row_loop < row+1; row_loop++)
        {
            for(col=0;col < row_loop; col++)
            {
                printf("*");
            }
            printf("\n");
        }

    /* Right inclined right angle triangle star pattern  */
    
        for(int row_loop = 0;row_loop < row+1; row_loop++)
        {
            for(int col = 0; col< row ; col++)
            {
                if(col >=  row-row_loop)
                printf("*");
                else
                printf(" ");
            }
            printf("\n");
        }
    
        /* Top Base Right inclined right angle triangle star pattern  */
        printf("\n");
        for(int row_loop = 0;row_loop < row+1; row_loop++)
        {
            for(int col = 0; col < row ; col++)
            {
                if(col <  row-row_loop)
                printf("*");
                else
                printf(" ");
            }
            printf("\n");
        }

        /* Top Base Right inclined right angle triangle star pattern  */
        for(int row_loop=0;row_loop < row+1; row_loop++)
        {
            for(col=0;col < row; col++)
            {
                if(col >= row_loop)
                    printf("*");
                else
                    printf(" ");
            }
            printf("\n");
        }

        printf("\r\nDo you want to run program again");
        scanf("%d",&run_again);
    }while(run_again);

    return 0;
}

File: rohit/js/star-pattern.js
function printLeftInclineRightAngleStar(ln) {
  for (let row = 1; row <= ln; row++) {
    let star = "";
    for (let col = 0; col < row; col++) {
      star += "*";
    }
    console.log(star);
  }
}

function printRightInclineStarPattern(line) {
  for (let row = 1; row <= line; row++) {
    let star = "";
    for (let col = 0; col < line; col++) {
      if (5 - row > col) {
        star += "-";
      } else {
        star += "*";
      }
    }

    console.log(star);
  }
}

function printReverseLeftInclineStarPattern(line) {
  for (let row = 1; row <= line; row++) {
    let star = "";
    for (let col = 0; col < line; col++) {
      if (5 - row < col) {
        star += "-";
      } else {
        star += "*";
      }
    }

    console.log(star);
  }
}

function printReverseRighttInclineRightAngleStar(ln) {
  for (let row = 0; row < ln; row++) {
    let star = "";
    for (let col = 0; col < ln; col++) {
      if (col >= row) {
        star += "*";
      } else {
        star += "-";
      }
    }
    console.log(star);
  }
}

function printStarPattern(line) {
  printLeftInclineRightAngleStar(line);
  printReverseLeftInclineStarPattern(line);
  printRightInclineStarPattern(line);
  printReverseRighttInclineRightAngleStar(line);
}

let line = prompt("enter line to print star pattern ", "");

printStarPattern(line);
